homeassistant/components/broadlink/cover.py

```diff
@@ -0,0 +1,545 @@
+"""Support for Broadlink RM device gateway for IR/RF covers."""
+from datetime import timedelta
+from ipaddress import ip_address
+import logging
+import socket
+
+import broadlink
+import voluptuous as vol
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    ATTR_TILT_POSITION,
+    ENTITY_ID_FORMAT,
+    PLATFORM_SCHEMA,
+    SUPPORT_CLOSE,
+    SUPPORT_CLOSE_TILT,
+    SUPPORT_OPEN,
+    SUPPORT_OPEN_TILT,
+    SUPPORT_SET_POSITION,
+    SUPPORT_SET_TILT_POSITION,
+    SUPPORT_STOP,
+    SUPPORT_STOP_TILT,
+    CoverDevice,
+)
+from homeassistant.const import (
+    CONF_COMMAND_CLOSE,
+    CONF_COMMAND_OPEN,
+    CONF_COMMAND_STOP,
+    CONF_COVERS,
+    CONF_FRIENDLY_NAME,
+    CONF_HOST,
+    CONF_MAC,
+    CONF_TIMEOUT,
+)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.event import track_utc_time_change
+from homeassistant.util import slugify
+from homeassistant.util.dt import utcnow
+
+from . import async_setup_service, data_packet, hostname, mac_address
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_OPENING_TIME = "opening_time"
+CONF_CLOSING_TIME = "closing_time"
+CONF_START_POS = "start_pos"
+
+CONF_TILT_COMMAND_OPEN = "tilt_command_open"
+CONF_TILT_COMMAND_CLOSE = "tilt_command_close"
+CONF_TILT_COMMAND_STOP = "tilt_command_stop"
+CONF_TILT_OPENING_TIME = "tilt_opening_time"
+CONF_TILT_CLOSING_TIME = "tilt_closing_time"
+CONF_TILT_START_POS = "tilt_start_pos"
+
+DEVICE_DEFAULT_NAME = "Broadlink cover"
+DEFAULT_TIMEOUT = 5
+DEFAULT_RETRY = 3
+DEFAULT_PORT = 80
+DEVICE_SIMULATION = False
+
+DEFAULT_TRAVEL_TIME = 2
+MAX_TRAVEL_TIME = 120
+POSITION_MIN = 0
+POSITION_MAX = 100
+DEFAULT_START_POS = 0
+
+TRAVEL_TIME = vol.All(vol.Coerce(float), vol.Range(min=0, max=MAX_TRAVEL_TIME))
+
+COVER_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_COMMAND_OPEN): data_packet,
+        vol.Optional(CONF_COMMAND_CLOSE): data_packet,
+        vol.Optional(CONF_COMMAND_STOP): data_packet,
+        vol.Optional(CONF_OPENING_TIME, default=0.0): TRAVEL_TIME,
+        vol.Optional(CONF_CLOSING_TIME, default=0.0): TRAVEL_TIME,
+        vol.Optional(CONF_START_POS, default=DEFAULT_START_POS): cv.positive_int,
+        vol.Optional(CONF_TILT_COMMAND_OPEN): data_packet,
+        vol.Optional(CONF_TILT_COMMAND_CLOSE): data_packet,
+        vol.Optional(CONF_TILT_COMMAND_STOP): data_packet,
+        vol.Optional(CONF_TILT_OPENING_TIME, default=0.0): TRAVEL_TIME,
+        vol.Optional(CONF_TILT_CLOSING_TIME, default=0.0): TRAVEL_TIME,
+        vol.Optional(CONF_TILT_START_POS, default=DEFAULT_START_POS): cv.positive_int,
+        vol.Optional(CONF_FRIENDLY_NAME): cv.string,
+    }
+)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

This integration needs to be refactored to use a config flow and config entries.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

Reply:
I am working on the refactor. Please hold this PR a little longer so we can integrate platforms.

Reply:
When submitted last December this PR purpose was for a platform addition to the Broadlink integration. With this April 14 ADR10, my understanding is that the scope of this PR needs to be extended to a complete refactoring of the entire Broadlink integration to make it an UI Integration.
@felipediel is it what you meant by refactor.

Reply:
@embak Home Assistant has changed and we need to adapt the Broadlink integration to be configured via config flow. I'm working on the update. When I'm done I'll be back here to help you adapt the code.

Reply:
This PR will be refactored pending config flow implementation in the Broadlink integration. ( #36914 )

Reply:
Can we in meantime close this PR and re-open when ready?

As we keep looking at things, that cannot move forward at this point, which is highly inefficient for us reviewers.

Thanks for considering 👍